src/System.Device.Gpio/System/Device/Gpio/ExtendedPinMode.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Device.Gpio
+{
+    /// <summary>
+    /// Platform-dependent description of an alternate pin mode (i.e. a pin that is used for I2C or SPI transfer).
+    /// Depending on the hardware in use, the use will reflect the effective mode ("PWM", "SPI") or the state of the
+    /// mode multiplexer of the CPU ("ALT0", "ALT1").
+    /// </summary>
+    public class ExtendedPinMode : IEquatable<ExtendedPinMode>
```

Comment:
I'd refrain from doing IEquatable for now unless we have a clear use case.

Reply:
The user needs to do comparisons (i.e. `if (mode == Modes.GPIO)...`). Without implementing IEquatable, things get needlessly complicated.

Reply:
This will still work considering you return the same reference

Reply:
You are right, that would work with reference comparison as well. But considering that the class is inheritable, it should be made clear that instances of it are meant to be compared against each other. Maybe some derived implementations will not use statics, but dynamically create the instances (i.e. because the set of possible modes is only known at runtime).